### PR TITLE
Making the sidebar inspector's tabs stick when scrolling.

### PR DESCRIPTION
**Pull request by shaunandrews**
When the editor's inspector sidebar is open the tabs will now remain at the top while scrolling the sidebar.

Sticky Tabs

This comes from (and resolves) #27873

### CODE REVIEW

File: packages/edit-post/src/components/sidebar/style.scss
Comment (jasmussen) [inline packages/edit-post/src/components/sidebar/style.scss:8]: Should this use the z-index mixin?

**Review by jasmussen — APPROVED** [review]
We used to have this, and it somehow regressed and disappeared in one of the skeleton refactors. So I'm happy to see it come back.

But I also recall there being issues with this at some breakpoints or other, and that was one of the reasons it was never resurfaced.

In my testing, I couldn't break it. I tried fullscreen and not fullscreen, I tried all our breakpoints in both, and it felt and behaved solidly to me. I think this needs a good testing in Firefox and potentially IE, and let's keep an eye on adjacent issues that might pop up, but this is awesome. Thank you Shaun.

**Comment by shaunandrews** [comment]
I started testing on mobile devices and realized that the tabs were already "sticky" on small viewports. I circled back and found a bunch of conflicting styles applied on different viewports. I removed those unneeded (I think?) styles and now the tabs are sticky everywhere.

I've tested on latest Chrome, Safari, Mobile Safari, Firefox, and Edge and all seems well.

Would love more eyes to make sure I'm not breaking some other use of the `complementary-area` component.

**Comment by jasmussen** [comment]
I definitely recall there being some junk, so thanks so much for the cleanup, it was much needed. 

And I'm still a fan of the sticky toolbar, though the recent change may have had an unintended side effect. Before:

before

After:

after

This bit, in particular:

Screenshot 2021-01-07 at 10 22 13

I know, everyone hates those scrollbars. But they are default on most platforms, so we have to design for them. I think I recall that when the tabs were stickied, the icons would align vertically. 

Can we keep most of the cleanup, but go back to when it was sticky so the icons line up?

**Comment by shaunandrews** [comment]
> they are default on most platforms, so we have to design for them.

Agreed. I did toggle the scrollbars on-and-off but didn't consider the icon alignment as unexpected. When you collapse the accordions, the icons line up. If you need to scroll, then the scrollbar pushes them over — but not the close (x) button, because it's not part of the scrollable area. The logic tracks, but I can understand the want to improve the visual balance at all times.

I'm not sure how to go about detecting the presence of a scrollbar (JS could do it, I'm sure), but I'll try to tinker some more and see what I can do.

**Comment by jasmussen** [comment]
> I'm not sure how to go about detecting the presence of a scrollbar (JS could do it, I'm sure), but I'll try to tinker some more and see what I can do.

Oh, please let's avoid JS! My point is, you had it working when you used position: sticky:

screencast 2021-01-08 10-09-55

I appreciate all the cleanup you did, but if you can rewind a few steps back and keep the sticky bit, it seems like that might thread the needle?

**Comment by shaunandrews** [comment]
> I appreciate all the cleanup you did, but if you can rewind a few steps back and keep the sticky bit, it seems like that might thread the needle?

Yes, of course. I've got it mostly working (just pushed an update to this PR), but now its broken on mobile. I'll circle back soon.

**Comment by shaunandrews** [comment]
Ok, I've got this working well on smaller screen. This _should_ be ready for another review.

File: packages/interface/src/components/complementary-area/style.scss
Comment (jasmussen) [inline packages/interface/src/components/complementary-area/style.scss:20]: Nice, using the mixin.

File: packages/interface/src/components/complementary-area/style.scss
Comment (jasmussen) [inline packages/interface/src/components/complementary-area/style.scss:23]: This could be `$panel-header-height`.

File: packages/interface/src/components/complementary-area/style.scss
Comment (jasmussen) [inline packages/interface/src/components/complementary-area/style.scss:25]: The small stuff appears to be working fine, so I ask only out of curiosity — why was this changed from small to medium?

**Review by jasmussen — APPROVED** [review]
Screenshot 2021-01-12 at 08 10 46

Digging it. I like all the red. I left a few comments, one that could use a fix, but otherwise, let's try this!

**Review by youknowriad — COMMENTED** [review]
Did we test how this impacts the other screens edit-site, edit-widgets and edit-navigation (just making sure we're not regressing there)

File: packages/interface/src/components/complementary-area/style.scss
Comment (shaunandrews) [reply packages/interface/src/components/complementary-area/style.scss:23]: Nice catch.

File: packages/interface/src/components/complementary-area/style.scss
Comment (shaunandrews) [reply packages/interface/src/components/complementary-area/style.scss:25]: Its a different set of declarations. Its not that I changed a small to a medium, but more that I removed a small, and added a medium.

**Comment by shaunandrews** [comment]
> Did we test how this impacts the other screens edit-site, edit-widgets and edit-navigation (just making sure we're not regressing there)

`edit-site` doesn't use tabs for its sidebar, so no changes there.

Global Styles does have tabs, but this change doesn't affect that sidebar.

`edit-widgets` tabs are sticky and working fine.

`edit-navigation` doesn't use tabs, as its "sidebar" is actually inside a popover.

**Comment by shaunandrews** [comment]
I've no idea what I did wrong, but GH just automatically requested a review from a bunch of code owners. While I appreciate any and all reviews, I also apologize for whatever I did to ping a bunch of people :)

**Comment by shaunandrews** [comment]
I tried to rebase, and just really messed this all up. Hopefully I can get it sorted tomorrow.

**Comment by gziolo** [comment]
There is a section in the contributor docs that might help: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/git-workflow.md#keeping-your-branch-up-to-date

**Comment by jasmussen** [comment]
Thank you Shaun for staying with this, and thank you @jeyip for fixing things up. I'll let Shaun do the honors and merge. But it should be ready!

**Comment by shaunandrews** [comment]
Its been quite the journey. Thanks for the reviews and help everyone.

**Pull request merged**
